### ISSUE TITLE
colorization doesn't work in large cases

### ISSUE THREAD

**Issue by povar81** [robotframework/RIDE#2037]
With smaller cases, up to 100 lines, it works fine, but with those that are 100 lines and more I get this situation:

Only the upper 100 lines are colored, the rest are not.

**Comment by povar81** [comment]
https://imgur.com/8vSka85

**Comment by Nyral** [comment]
```
  File "C:\Python37\lib\site-packages\robotide\editor\gridcolorizer.py", line 50, in _colorize_cell
    cell_info = self._controller.get_cell_info(row, col)
  File "C:\Python37\lib\site-packages\robotide\controller\macrocontrollers.py", line 174, in get_cell_info
    return steps[row].get_cell_info(col)
  File "C:\Python37\lib\site-packages\robotide\controller\stepcontrollers.py", line 85, in get_cell_info
    content = self._get_content_with_type(col, position)
  File "C:\Python37\lib\site-packages\robotide\controller\stepcontrollers.py", line 163, in _get_content_with_type
    if self.is_user_keyword(value):
  File "C:\Python37\lib\site-packages\robotide\controller\stepcontrollers.py", line 202, in is_user_keyword
    return self.parent.is_user_keyword(value)
  File "C:\Python37\lib\site-packages\robotide\controller\macrocontrollers.py", line 180, in is_user_keyword
    return self.datafile_controller.is_user_keyword(value)
  File "C:\Python37\lib\site-packages\robotide\controller\filecontrollers.py", line 207, in is_user_keyword
    return WithNamespace.is_user_keyword(self, self.datafile, value)
  File "C:\Python37\lib\site-packages\robotide\controller\basecontroller.py", line 99, in is_user_keyword
    return self._namespace.is_user_keyword(datafile, value)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 194, in is_user_keyword
    return bool(self.find_user_keyword(datafile, kw_name))
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 190, in find_user_keyword
    kw = self.find_keyword(datafile, kw_name)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 212, in find_keyword
    self._context_factory)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 517, in get_keywords_cached
    datafile, context_factory.ctx_for_datafile(datafile))
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 416, in get_keywords_from
    self._get_vars_recursive(datafile, ctx)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 488, in _get_vars_recursive
    ctx.set_variables_from_datafile_variable_table(datafile)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 248, in set_variables_from_datafile_variable_table
    self.vars.set_from_variable_table(datafile.variable_table)
  File "C:\Python37\lib\site-packages\robotide\namespace\namespace.py", line 333, in set_from_variable_table
    for variable in variable_table:
  File "C:\Python37\lib\site-packages\robotide\lib\robot\parsing\model.py", line 482, in __iter__
    return iter(self.variables)
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by povar81** [comment]
But it used to work fine on older versions (1.5.2.1 for instance, and older).
I have cases with 150-200 lines and they are displayed normally on older versions.

**Comment by HelioGuilherme66** [comment]
@povar81 Like I mentioned on other issue, please provide more information on the versions you are using. In this case, it was working on 1.5.2.1. Was that with wxPython 2.8.12.1? There was big changes in wxPython since then. Also the library responsible for colorization is Pygments, maybe related? On the other end, the large test cases you are using, maybe using too much memory, and so those problems are appearing.

**Comment by povar81** [comment]
ok, the new ride with the colorization problem:
Windows 7.
Python 3.7.4,
robotframework==3.1.2,
robotframework-ride==1.7.4b1.dev1 (I'm usring dev1 version because it's the last one supporting the old for loop syntax)
wxPython==4.0.6

the old one working normally even with 370 lines:
RIDE 1.5.2.1 running on Python 2.7.12. 
robotframework==2.8.5
>>> wx.version()
'2.8.12.1 (msw-unicode)'

**Comment by povar81** [comment]
Fixed in version RIDE 1.7.4rc1.dev on RIDE 1.7.4rc1.dev (Windows 7, Python 3.7.4). 

In fact it works even better now, because on older versions (1.5.2.1 and older) when I had even large cases with over 200+ lines it just 'hanged' and after some time I had to kill the process in taskmanager.
